src/server/qgsserviceregistry.cpp

```diff
@@ -0,0 +1,166 @@
+/***************************************************************************
+                          qgsserviceregistry.cpp
+
+  Class defining the service manager for QGIS server services.
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsserviceregistry.h"
+#include "qgsservice.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+
+namespace {
+
+// Build a key entry from name and version
+QString makeServiceKey( const QString& name, const QString& version  )
+{
+  return QString( "%1_%2" ).arg(name,version);
+}
+
+// Compare two version strings:
+// The strings are splitted into dot separated segment
+// Each segment are compared up to the shortest number of segment of the
+// lists. Remaining segments are dropped.
+// If both segments can be intepreted as numbers the are compared as numbers, otherwise
+// They are compared lexicographically.
+// Return true if v1 is greater than v2
+bool isVersionGreater( const QString& v1, const QString& v2 )
```

Comment:
I've been thinking about this method - I think this could be moved to a static member in QgsProjectVersion, and then QgsProjectVersion renamed to something generic like QgsVersionNumber. It's quite a handy method which has uses outside of server too.